### PR TITLE
[GH-13] Set bad words list from Mattermost settings UI

### PR DESCRIPTION
**Pull request by nathanaelhoun**
#### Summary

- [x] Bad words list is accessible from Mattermost settings UI
- [x] Settings are protected from concurrent access

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-profanity-filter/issues/13

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -25,9 +25,16 @@
...
       }
     ],
     "header": "",
     "footer": ""
   }
-}
+}
```
Comment (hanzei) [inline plugin.json:40]: Nit pick: Please add a newline

Comment (nathanaelhoun) [reply plugin.json]: Done :white_check_mark:

**Comment by nathanaelhoun** [comment]
Thanks !
I have tried to write a few tests for this changes (these are my first tests), but I am not sure if they are relevant.

In `configuration_test.go` :
```go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestConfiguration(t *testing.T) {
	t.Run("null configuration", func(t *testing.T) {
		plugin := &Plugin{}
		assert.NotNil(t, plugin.getConfiguration())
	})

	t.Run("changing configuration", func(t *testing.T) {
		plugin := &Plugin{}

		configuration1 := &configuration{RejectPosts: false}
		plugin.setConfiguration(configuration1)
		assert.Equal(t, configuration1, plugin.getConfiguration())

		configuration2 := &configuration{RejectPosts: true}
		plugin.setConfiguration(configuration2)
		assert.Equal(t, configuration2, plugin.getConfiguration())
		assert.NotEqual(t, configuration1, plugin.getConfiguration())
		assert.False(t, plugin.getConfiguration() == configuration1)
		assert.True(t, plugin.getConfiguration() == configuration2)
	})

	t.Run("setting same configuration", func(t *testing.T) {
		plugin := &Plugin{}

		configuration1 := &configuration{}
		plugin.setConfiguration(configuration1)
		assert.Panics(t, func() {
			plugin.setConfiguration(configuration1)
		})
	})
}
```

Is this a good idea and useful ?

**Comment by hanzei** [comment]
In general having tests is always helps. The most impact have the ones that actually test business logic. In this case, having a test that ensures `p.badWords` gets changed when ever the configuration changes, would be awesome.

**Comment by hanzei** [comment]
@DHaussermann Skipping QA testing for this no-marketplace plugin

**Comment by NewFlocon** [comment]
Oh that looks really good ! Thank you a lot @nathanaelhoun !

**Pull request merged**